packages/headless/src/app/headless-engine.ts

```diff
@@ -284,12 +281,12 @@ export class HeadlessEngine<Reducers extends ReducersMapObject>
         },
       }),
     });
-    validateOptions(
-      this,
-      configurationSchema,
-      options.configuration,
-      HeadlessEngine.name
-    );
+    try {
+      configurationSchema.validate(options.configuration);
+    } catch (error) {
+      this.logger.error(error, 'Headless engine configuration error');
```

Comment:
So, we * most probably * want to throw here because any error on the engine has a good chance of being "fatal", and the whole application won't work.

It's not 100% always the case, as some options being invalid could still mean the whole app could potentially still work correctly, but I think the generalization here is good.

We could have a small UT to ensure the engine is validated and error is thrown back.